stablehlo/dialect/TypeInference.cpp

```diff
@@ -692,10 +692,10 @@ LogicalResult inferConcatenateOp(Optional<Location> location, ValueRange inputs,
   return success();
 }
 
-LogicalResult inferCreateTokenOp(MLIRContext* context,
-                                 Optional<Location> location,
+LogicalResult inferCreateTokenOp(Dialect* dialect, Optional<Location> location,
                                  SmallVectorImpl<Type>& inferredReturnTypes) {
-  inferredReturnTypes.push_back(stablehlo::TokenType::get(context));
+  auto hloDialect = cast<HloDialectInterface>(dialect);
```

Comment:
Not sure if there's much of an advantage, but could use `dialect->getRegisteredInterface<HloDialectInterface>()` I believe. Looks like it was made for retrieving dialect interfaces, not sure if its outdated or preferred.

https://mlir.llvm.org/doxygen/classmlir_1_1Dialect.html#a46a6a74902ee28426cebc490d04ee67b

Reply:
Looks like `cast` [calls](https://github.com/llvm/llvm-project/blob/b4028fbc1a8853c34517f44ef779d517a44bd71d/mlir/include/mlir/IR/Dialect.h#L324-L339) `getRegisteredInterface`. I think let's stick to `cast` which is uniform with other forms of casting that we have in the codebase.